lib/keyverificationsession.cpp

```diff
@@ -351,9 +351,8 @@ void KeyVerificationSession::handleStart(const KeyVerificationStartEvent& event)
     const auto pubkeyLength = olm_sas_pubkey_length(olmData);
     auto publicKey = byteArrayForOlm(pubkeyLength);
     olm_sas_get_pubkey(olmData, publicKey.data(), pubkeyLength);
-    const auto canonicalEvent = QString(QJsonDocument(event.contentJson()).toJson(QJsonDocument::Compact));
-    auto commitment = QString(QCryptographicHash::hash((QString(publicKey) % canonicalEvent).toLatin1(), QCryptographicHash::Sha256).toBase64());
-    commitment = commitment.left(commitment.indexOf('='));
+    const auto canonicalEvent = QJsonDocument(event.contentJson()).toJson(QJsonDocument::Compact);
+    auto commitment = QCryptographicHash::hash(publicKey + canonicalEvent, QCryptographicHash::Sha256).toBase64(QByteArray::OmitTrailingEquals);
```

Comment:
I hunted after conversions too aggressively in the previous suggestion.
```suggestion
    const auto commitment =
        QString::fromLatin1(QCryptographicHash::hash(publicKey + canonicalEvent,
                                                     QCryptographicHash::Sha256)
                                .toBase64(QByteArray::OmitTrailingEquals));
```